### PR TITLE
[toplevel] allow toploop_init change Coq options

### PR DESCRIPTION
**Pull request by gares**
The API is not as expressive as the old global `Flags`.
I can hack around this in pidetop 8.8, hence the milestone.

### CODE REVIEW

**Comment by ejgallego** [comment]
Maybe you could add PIDEtiop to the CI, if you think the project is still worth maintaining?

File: toplevel/coqtop.ml [hunk 1]
```diff
@@ -435,14 +435,26 @@ let init_toplevel arglist =
...
+        exit (exitcode opts)
+      end;
+      if opts.print_tags then begin
+        print_style_tags opts;
+        exit (exitcode opts)
+      end;
+      if opts.filter_opts then begin
+        print_string (String.concat "\n" extras);
+        exit 0;
+      end;
```
Comment (ejgallego) [inline toplevel/coqtop.ml]: I am sorry this change seems unrelated.

**Review by ejgallego — CHANGES_REQUESTED** [review]
The change is fine and indeed makes sense, however remove the unrelated changes. As you need to push a new version, please extend the commit message documenting the options your toplevel modifies.

File: toplevel/coqtop.ml (hunk 1 above)
Comment (gares) [reply toplevel/coqtop.ml]: It is in a separate commit, do you want a separate PR?

**Comment by gares** [comment]
Not sure I understand what you want me to document. There is no such a thing as my top-level. The API lets you change any option, especially the ones with no command line flag.

**Comment by gares** [comment]
Maybe you meant "which flag does pidetop needs to change"?
A couple is STM, like the never_reopen_branch one.

**Comment by ejgallego** [comment]
> A couple is STM, like the never_reopen_branch one.

Yup, that's what I meant, you could write a more informative commit message such as:
> [toplevel] allow toploop_init change Coq options
>
> Toplevels may want to modify for example the `Stm` flags, which after #1108 are handled in a functional way.

Also, please change the title of the PR to reflect the above.

With regards to the indentation commit, I'd much prefer to see it discussed in a separate PR if possible, as I don't think the new indentation improves on the old one.

**Comment by ejgallego** [comment]
No need to bother Pierre-Marie with this one-liner.

**Comment by gares** [comment]
@ejgallego done

**Review by ejgallego — APPROVED** [review]

**Comment by ejgallego** [comment]
Both CI failures are unrelated, merging as soon as the CI ends.
